### PR TITLE
Add threadpool

### PR DESCRIPTION
**Pull request by BitSyndicate1**
setup global threadpool

### CODE REVIEW

File: vr-threading/src/lib.rs [hunk 1]
```diff
@@ -1,14 +1,107 @@
...
+type Job = Box<dyn FnOnce() + Send + 'static>;
+
+pub struct StandardThreadPool {
+    sender: Option<mpsc::Sender<Job>>,
+    workers: Vec<Worker>,
+}
+
+pub mod global_threadpool;
+
+pub static mut THREADPOOL: StandardThreadPool = StandardThreadPool::new();
```
Comment (anonymous123-code) [inline vr-threading/src/lib.rs]: I don't think that the thread pool should be global.

File: vr-threading/src/lib.rs [hunk 2]
```diff
@@ -1,14 +1,107 @@
...
+}
+
+impl StandardThreadPool {
+    pub const fn new() -> Self {
+        StandardThreadPool {
+            sender: None,
+            workers: Vec::new(),
+        }
+    }
+    unsafe fn init(&mut self, size: NonZeroUsize) {
```
Comment (anonymous123-code) [inline vr-threading/src/lib.rs]: Is this really unsafe?

File: vr-threading/src/lib.rs [hunk 3]
```diff
@@ -1,14 +1,107 @@
...
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+        for worker in self.workers.iter_mut() {
+            println!("ThreadPool: Shutting down worker {}", worker.id);
+            if let Some(thread) = worker.thread.take() {
+                thread.join().unwrap();
+            }
+        }
     }
 }
```
Comment (anonymous123-code) [inline vr-threading/src/lib.rs]: Test pls

File: vr-threading/src/lib.rs [hunk 4]
```diff
@@ -1,14 +1,107 @@
...
+        drop(self.sender.take());
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+        for worker in self.workers.iter_mut() {
+            println!("ThreadPool: Shutting down worker {}", worker.id);
+            if let Some(thread) = worker.thread.take() {
+                thread.join().unwrap();
```
Comment (anonymous123-code) [inline vr-threading/src/lib.rs]: If join waits for all pools to finish, but a pool has no task, this might hold, I think.

**Review by anonymous123-code — CHANGES_REQUESTED** [review]

File: vr-threading/src/lib.rs (hunk 4 above)
Comment (BitSyndicate1) [reply vr-threading/src/lib.rs]: No it is fully valid.

**Comment by BitSyndicate1** [comment]
Not finished working out memory freeing problems

**Review by anonymous123-code — APPROVED** [review]

**Pull request merged**